### PR TITLE
Fixed old forum thread references

### PR DESCRIPTION
**Pull request by Rohaq**
I have too much free time on my hands, apparently.

### CODE REVIEW

**Comment by Rohaq** [comment]
Interesting, it's failing the build on the file firespitter-downloader - a Perl file, with only a basic string change as far as I can see. I'll have a dig around.

**Comment by Rohaq** [comment]
Ah, it's trying to run a JSON validator against a Perl file.

Yep, that'd cause issues.

**Comment by HebaruSan** [comment]
Yeah, the validation script is not picky enough concerning which files are and are not metadata. It lives here in case you feel like fixing that:

https://github.com/KSP-CKAN/xKAN-meta_testing/blob/master/NetKAN/build.sh

File: NetKAN/KlockheedMartian-Gimbal.netkan [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "author": [ "dtobi", "Sarbian" ],
     "license": "CC-BY-NC-ND-3.0",
     "resources": {
-        "homepage": "http://forum.kerbalspaceprogram.com/threads/55985",
+        "homepage": https://forum.kerbalspaceprogram.com/index.php?/topic/76049-090-on-hold-klockheed-martian-special-parts-v22-28-nov-dev-thread/",
```
[the remark this answers is not in the document]
Comment (Rohaq) [reply NetKAN/KlockheedMartian-Gimbal.netkan]: Yep, found that in the build output. Testing now :)

**Comment by HebaruSan** [comment]
The 403 forbidden errors are GitHub throttling. We can probably ignore those but it would be nice to get through a full pass cleanly.

**Comment by Rohaq** [comment]
Hmm, any idea how can we stop falling afoul of Github throttling?

**Comment by HebaruSan** [comment]
I was planning to wait an hour or two and force a rebuild.

**Comment by Rohaq** [comment]
Fair enough - I've just sent a PR for KSP-CKAN/xKAN-meta_testing to skip over validating files outside of the NetKAN directory too.

**Comment by HebaruSan** [comment]
#rebuild

**Comment by Rohaq** [comment]
It gets further, but the rate limit on Github is 60, even when it resets - I don't think this will ever pass :(

```
$ curl -i https://api.github.com/users/rohaq
...
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 56
X-RateLimit-Reset: 1523151078
...
```

**Comment by HebaruSan** [comment]
Well, the syntax checks all passed, and Olympic1 and I have verified visually that the changes only affect the forum URLs, so I think we can skip the full validation in this case. And it should run with an auth token once the web hook gets it, bumping that 60 limit to 5000. Merging...

**Comment by Rohaq** [comment]
Cheers, apologies for the long list of changes!

**Comment by Dazpoet** [comment]
It's like the second coming of @plague006 but all in a single PR O_o

**Comment by plague006** [comment]
Glad I'm not the only one who likes up to date forum threads! :D

**Comment by Rohaq** [comment]
I fixed one, and then decided to do a regex search for any others.

197! D:

It's just a shame that the URL will change whenever people edit the title. Stupid forum software :(

**Comment by HebaruSan** [comment]
I think everything after the first hyphen in the URL is basically ignored, and the part before that is the thread ID. Hence the practice of formatting them like this sometimes:

https://github.com/KSP-CKAN/NetKAN/blob/d7440ce2eb3dd171f789e8f1a0233e8bc7954fdf/NetKAN/PlanetShine.netkan#L11

**Comment by Rohaq** [comment]
Ah good, slightly less stupid forum software then!

**Pull request merged**